src/B0_data_input_json.py

```diff
@@ -48,32 +52,35 @@ def convert_special_types(a_dict, prev_key=None):
 
     """
 
+    answer = a_dict
     if isinstance(a_dict, dict):
-        # the a_dict argument is a dictionary, therefore we dive deeper in the nesting level
-        answer = {}
-        for k in a_dict:
-            answer[k] = convert_special_types(a_dict[k], prev_key=k)
-
-    else:
-        # the a_dict argument is not a dictionary, therefore we check if is one the serialized type
-        # pandas.Series, pandas.DatetimeIndex, pandas.DataFrame, numpy.array
-        answer = a_dict
-        if isinstance(a_dict, str):
-            if TYPE_DATAFRAME in a_dict:
-                a_dict = a_dict.replace(TYPE_DATAFRAME, "")
+        # the a_dict argument is a dictionary not containing the special type key,
+        # therefore we dive deeper in the nesting level
+        if DATA_TYPE_JSON_KEY not in a_dict:
+            answer = {}
+            for k in a_dict:
+                answer[k] = convert_from_json_to_special_types(a_dict[k], prev_key=k)
```

Comment:
I didn't know you could use 'answer[k]' to update a dict. Thought one had to use 'answer.update({})'.

Reply:
If you have only one key to update, I think the two are equivalent